core/src/main/java/org/elasticsearch/search/suggest/SuggestionBuilder.java

```diff
@@ -197,7 +197,18 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
 
     protected abstract SuggestionBuilder<T> innerFromXContent(QueryParseContext parseContext, String name) throws IOException;
 
-    protected abstract SuggestionContext build(QueryShardContext context) throws IOException;
+    public SuggestionContext build(QueryShardContext context) throws IOException {
+        SuggestionContext suggestionContext = innerBuild(context);
+        // copy over common settings to each suggestion builder
+        populateCommonFields(context.getMapperService(), suggestionContext);
```

Comment:
To keep things simple, I agree with @cbuescher. We can just call `populateCommonFields` from inside `innerBuild`/`build` to be explicit in ordering what params get set there? maybe make `populateCommonFields` protected and use it in the sub-classes? Sure, we will duplicate calling this in all the implementations, but it seems simpler than another `newInstance` abstract method?